go/database/mpt/io/verification_proof.go

```diff
@@ -262,104 +270,95 @@ func generateUnusedKeys(trie verifiableTrie, number int, address common.Address)
 	return res, nil
 }
 
-// accountVerifyingVisitor is a visitor that verifies the consistency of witness proofs for a live trie.
+// accountVerifyingProofVisitor is a visitor that verifies the consistency of witness proofs for a live trie.
 // It collects account and storage slots and extracts witness proofs for each account and its storage.
 // It checks that values in the database and the proof match.
 // The process can be interrupted by the input context.
-type accountVerifyingVisitor struct {
-	ctx      context.Context
-	rootHash common.Hash
-	trie     verifiableTrie
-	observer mpt.VerificationObserver
-
-	err error
+type accountVerifyingProofVisitor struct {
+	ctx       context.Context
+	rootHash  common.Hash
+	trie      verifiableTrie
+	observer  mpt.VerificationObserver
+	directory string
+	config    mpt.MptConfig
 
 	logWindow    int
 	counter      int
 	numAddresses int
 }
 
-func (v *accountVerifyingVisitor) Visit(n mpt.Node, _ mpt.NodeInfo) mpt.VisitResponse {
+func (v *accountVerifyingProofVisitor) Visit(n mpt.Node, nodeInfo mpt.NodeInfo) error {
 	if v.counter%100 == 0 && interrupt.IsCancelled(v.ctx) {
-		v.err = interrupt.ErrCanceled
-		return mpt.VisitResponseAbort
+		return interrupt.ErrCanceled
 	}
 	v.counter++
 
 	switch n := n.(type) {
 	case *mpt.AccountNode:
 		proof, err := v.trie.CreateWitnessProof(n.Address())
 		if err != nil {
-			v.err = err
-			return mpt.VisitResponseAbort
+			return err
 		}
 		if !proof.IsValid() {
-			v.err = ErrInvalidProof
-			return mpt.VisitResponseAbort
+			return ErrInvalidProof
 		}
 
-		if err := verifyAccount(v.rootHash, proof, n.Address(), n.Info()); err != nil {
-			v.err = err
-			return mpt.VisitResponseAbort
+		if err := verifyAccountProof(v.rootHash, proof, n.Address(), n.Info()); err != nil {
+			return err
 		}
 
-		storageVisitor := storageVerifyingVisitor{
+		storageVisitor := storageVerifyingProofVisitor{
 			ctx:            v.ctx,
 			rootHash:       v.rootHash,
 			trie:           v.trie,
 			currentAddress: n.Address(),
 			storage:        make(map[common.Key]common.Value)}
 
-		if err := v.trie.VisitAccountStorage(n.Address(), &storageVisitor); err != nil || storageVisitor.err != nil {
-			v.err = errors.Join(err, storageVisitor.err)
-			return mpt.VisitResponseAbort
+		// visit the storage of the account
+		if err := visitAll(v.directory, v.config, nodeInfo.Id, &storageVisitor, false); err != nil {
```

Comment:
If I recall correctly, this will visit all nodes of the trie, passing through all storage slots of all accounts. This means the total runtime of this should be something like `O(#accounts * #total_slots)`, which should be be quite extensive. Is this still running in an acceptable time?